packages/big-design/src/components/TableNext/Row/Row.tsx

```diff
@@ -1,60 +1,165 @@
-import { DragIndicatorIcon } from '@bigcommerce/big-design-icons';
-import React, { forwardRef, TableHTMLAttributes } from 'react';
+import { ChevronRightIcon, DragIndicatorIcon, ExpandMoreIcon } from '@bigcommerce/big-design-icons';
+import React, { forwardRef, ReactNode, TableHTMLAttributes } from 'react';
 
 import { typedMemo } from '../../../utils';
+import { Box } from '../../Box';
 import { Checkbox } from '../../Checkbox';
 import { DataCell } from '../DataCell';
-import { TableColumn, TableItem } from '../types';
+import { TableColumn, TableItem, TableSelectable } from '../types';
 
-import { StyledTableRow } from './styled';
+import { StyledExpandedIcon, StyledTableRow } from './styled';
+import { useRowState } from './useRowState';
+
+interface PrivateProps {
+  forwardedRef?: React.Ref<HTMLTableRowElement>;
+}
 
 export interface RowProps<T> extends TableHTMLAttributes<HTMLTableRowElement> {
+  childrenRows: T[];
+  childRowIndex?: number;
   columns: Array<TableColumn<T>>;
   headerCellWidths: Array<number | string>;
-  item: T;
+  isDraggable: boolean;
   isDragging?: boolean;
+  isExpanded?: boolean;
+  isExpandable?: boolean;
+  isParentRow?: boolean;
   isSelected?: boolean;
   isSelectable?: boolean;
+  item: T;
+  parentRowIndex: number;
+  selectedItems: TableSelectable['selectedItems'];
   showDragIcon?: boolean;
-  onItemSelect?(item: T): void;
-}
-
-interface PrivateProps {
-  forwardedRef?: React.Ref<HTMLTableRowElement>;
+  onExpandedRow?(parentRowIndex: number | null): void;
+  onItemSelect?(
+    childRowIndex: number | null,
+    index: number,
+    childrenRows: T[],
+    isParentRow: boolean,
+    isExpandable: boolean,
```

Comment:
Is there a way where we can reduce the amount of arguments passed? Anything greater than 2-3 arguments can be quite cumbersome. Could this be a single interface a pass in instead (like it was before)?